test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -570,6 +572,169 @@ def test_add_verbatim_box_result_types():
+        .add_instruction(Instruction(Gate.CNot(), [1, 2]))
+        .add_instruction(Instruction(Gate.CNot(), [2, 3]))
+        .add_instruction(Instruction(Measure(), 0))
+        .add_instruction(Instruction(Measure(), 1))
+        .add_instruction(Instruction(Measure(), 3))
+    )
+    assert circ == expected
+    assert circ._measure_targets == [0, 1, 3]
+
+
+def test_measure_with_noise():
+    circ = Circuit().x(0).x(1).bit_flip(0, probability=0.1).measure(0)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.X(), 0))
+        .add_instruction(Instruction(Gate.X(), 1))
+        .add_instruction(Instruction(BitFlip(probability=0.1), 0))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    assert circ == expected
+
+
+def test_measure_verbatim_box():
+    circ = Circuit().add_verbatim_box(Circuit().x(0).x(1)).measure(0)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(compiler_directives.StartVerbatimBox()))
+        .add_instruction(Instruction(Gate.X(), 0))
+        .add_instruction(Instruction(Gate.X(), 1))
+        .add_instruction(Instruction(compiler_directives.EndVerbatimBox()))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    expected_ir = OpenQasmProgram(
+        source="\n".join(
+            [
+                "OPENQASM 3.0;",
+                "bit[1] b;",
+                "qubit[2] q;",
+                "#pragma braket verbatim",
+                "box{",
+                "x q[0];",
+                "x q[1];",
+                "}",
+                "b[0] = measure q[0];",
+            ]
+        ),
+        inputs={},
+    )
+    assert circ == expected
+    assert circ.to_ir("OPENQASM") == expected_ir
+
+
+def test_measure_in_verbatim_subcircuit():
+    message = "Verbatim subcircuit is not measured."
+    with pytest.raises(ValueError, match=message):
+        Circuit().add_verbatim_box(Circuit().x(0).x(1).measure(0))
+
+
+def test_measure_qubits_out_of_range():
+    with pytest.raises(IndexError):
+        Circuit().h(0).cnot(0, 1).measure([4])
+
+
+def test_measure_empty_circuit():
+    with pytest.raises(IndexError):
+        Circuit().measure()
+
+
+def test_measure_no_target():
+    message = "Measure must include one or more target qubits."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).cnot(0, 1).measure()
+
+
+def test_measure_with_result_types():
+    message = "Cannot perform a measure instruction with a result type."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).sample(observable=Observable.Z(), target=0).measure(0)
+
+
+def test_result_type_with_measure():
+    message = "Cannot add a result type with a measure instruction."
+    with pytest.raises(ValueError, match=message):
+        Circuit().h(0).measure(0).sample(observable=Observable.Z(), target=0)
+
+
+def test_measure_with_multiple_measures():
+    message = "Cannot perform more than one measure instruction."
```

Comment:
this makes sense, but I'm not sure if there should be an issue or it just combines the targets. Another consideration, should we think about not allowing gates after measure instructions?

For both of these considerations, I think it's important to think about subroutines. For example what if a circuit subroutine uses 5 qubits, but only needs to measure 2, and you want to make a circuit that runs this subroutine on 3 sets of 5 qubits and get a total of 6 qubits measured

Reply:
I think we should not allow gates after measure instructions. That would likely cause confusion as it would not perform mid-circuit measurement. I have updated `circuit.py` to raise an error if a gate is added after a measure instruction.

We should allow multiple measure instructions so long as they are at the end of the circuit. The `braket_program_context` will treat each individual OpenQASM `measure` as a separate instruction that will lead to multiple measure instructions that we should combine at the end of the circuit.

This circuit `Circuit().h(0).cnot(0, 1).cnot(1, 2).cnot(2, 3).measure([0, 2]).measure(3).measure(1)` would measure the targets at the end but the diagram should probably be cleaned up to have all the measurements lined up regardless of each one's target(s).
```
T  : │  0  │  1  │  2  │  3  │     4     │
      ┌───┐                   ┌───┐       
q0 : ─┤ H ├───●───────────────┤ M ├───────
      └───┘   │               └─┬─┘       
            ┌─┴─┐               │   ┌───┐ 
q1 : ───────┤ X ├───●───────────┼───┤ M ├─
            └───┘   │           │   └───┘ 
                  ┌─┴─┐       ┌─┴─┐       
q2 : ─────────────┤ X ├───●───┤ M ├───────
                  └───┘   │   └───┘       
                        ┌─┴─┐ ┌───┐       
q3 : ───────────────────┤ X ├─┤ M ├───────
                        └───┘ └───┘       
T  : │  0  │  1  │  2  │  3  │     4     │
```

Reply:
Now that measure is treated as a single qubit instruction, this diagram for `Circuit().h(0).cnot(0, 1).cnot(1, 2).cnot(2, 3).measure([0, 2]).measure(3).measure(1)` looks cleaner as:

```
        T  : │  0  │  1  │  2  │  3  │  4  │
              ┌───┐                   ┌───┐ 
        q0 : ─┤ H ├───●───────────────┤ M ├─
              └───┘   │               └───┘ 
                    ┌─┴─┐             ┌───┐ 
        q1 : ───────┤ X ├───●─────────┤ M ├─
                    └───┘   │         └───┘ 
                          ┌─┴─┐       ┌───┐ 
        q2 : ─────────────┤ X ├───●───┤ M ├─
                          └───┘   │   └───┘ 
                                ┌─┴─┐ ┌───┐ 
        q3 : ───────────────────┤ X ├─┤ M ├─
                                └───┘ └───┘ 
        T  : │  0  │  1  │  2  │  3  │  4  │
```

similar to how `Circuit.h([0, 1])` is formatted
```
T  : │  0  │
      ┌───┐ 
q0 : ─┤ H ├─
      └───┘ 
      ┌───┐ 
q1 : ─┤ H ├─
      └───┘ 
T  : │  0  │
```

Reply:
Ya I agree with your call here, looks good to me